### PR TITLE
this commit includes completed ticket 9 including error handling

### PR DESCRIPTION
**Pull request by ccccc170**
this pull request includes completed ticket 9 with error handling

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -164,6 +164,54 @@ describe("GET /api/articles/:article_id", () => {
...
+      return request(app)
+        .get(`/api/articles/${articleId}/comments`)
+        .expect(200)
+        .then(({ body }) => {
+          expect(body).toBeInstanceOf(Array);
+          expect(body).toHaveLength(11);
+          body.forEach((user) => {
+            expect(typeof user).toBe("object");
+            expect(user !== null).toBe(true);
+            expect(!Array.isArray(user)).toBe(true);
```
Comment (karlaferry) [inline __tests__/app.test.js:180]: Not necessary to check if it's not an array when we already checked the type in line 178.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -164,6 +164,54 @@ describe("GET /api/articles/:article_id", () => {
...
+      return request(app)
+        .get("/api/articles/not-an-id/comments")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("bad request: invalid id!");
+        });
+    });
+  });
+});
+
```
Comment (karlaferry) [inline __tests__/app.test.js:214]: What if the article is valid, but it doesn't exist in the database? What if it does exist, but it doesn't have any comments?

**Review by karlaferry — CHANGES_REQUESTED** [review]
Almost there! Just a couple of things!

File: __tests__/app.test.js (hunk 2 above)
Comment (ccccc170) [reply __tests__/app.test.js:214]: Hi Karla, I am unsure about the situation where an article could be valid but not in the database. If the artticle is valid, how would it not be in the database?

File: __tests__/app.test.js (hunk 1 above)
Comment (ccccc170) [reply __tests__/app.test.js:180]: Hi, I have used this test in all prior tests so far. I did always feel it was perhaps a little too thorough but it was my understanding that arrays and null will be classed as an "object" when doing typeof, is this not the case or is it still not worth checking for that it isn't an array?

**Pull request closed without merging**